lib/elixir/lib/calendar.ex

```diff
@@ -1214,13 +1436,11 @@ defmodule NaiveDateTime do
   def to_iso8601(naive_datetime)
 
   def to_iso8601(%{year: year, month: month, day: day,
-                   hour: hour, minute: minute, second: second, microsecond: microsecond, calendar: Calendar.ISO}) do
-    Calendar.ISO.naive_datetime_to_iso8601(year, month, day, hour, minute, second, microsecond)
-  end
-
-  def to_iso8601(%{year: _, month: _, day: _,
-                   hour: _, minute: _, second: _, microsecond: _, calendar: _} = naive_datetime) do
-    raise ArgumentError, "cannot convert #{inspect naive_datetime} to the ISO 8601 format, because it does not use Calendar.ISO"
+                   hour: hour, minute: minute, second: second, microsecond: microsecond, calendar: calendar} = naive_datetime) do
+    utc_datetime =
+      naive_datetime
+      |> convert!(Calendar.ISO)
+    Calendar.ISO.naive_datetime_to_iso8601(utc_datetime.year, utc_datetime.month, utc_datetime.day, utc_datetime.hour, utc_datetime.minute, utc_datetime.second, utc_datetime.microsecond)
```

Comment:
Please use pattern matching to extract the values instead of `utc_datetime.field`.